src/openboardview/FileFormats/XZZPCBFile.cpp

```diff
@@ -0,0 +1,890 @@
+
+	des_decrypt(buf);
+
+	uint32_t current_pointer = 0;
+	uint32_t part_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+	current_pointer += 4;
+	current_pointer += 18;
+	uint32_t part_group_name_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+	current_pointer += 4;
+	current_pointer += part_group_name_size;
+	
+	// So far 0x06 sub blocks have been first always
+	// Also contains part name so needed before pins
+	if (buf[current_pointer] != 0x06) {
+		exit(1);
+	}
+	
+	current_pointer += 31;
+	uint32_t part_name_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+	current_pointer += 4;
+	std::string part_name(reinterpret_cast<char*>(&buf[current_pointer]), part_name_size);
+	current_pointer += part_name_size;
+
+	part.name = strdup(part_name.c_str());
+	part.mounting_side = BRDPartMountingSide::Top;
+	part.part_type = BRDPartType::SMD;
+
+	uint32_t pin_count = 0;
+	while (current_pointer <= part_size) {
+		uint8_t sub_type_identifier = buf[current_pointer];
+		current_pointer += 1;
+
+		switch (sub_type_identifier) {
+			case 0x01: {
+				// Currently unsure what this is
+				current_pointer += *reinterpret_cast<uint32_t*>(&buf[current_pointer]) + 4; // Skip the block
+				break;
+			}
+			case 0x05: { // Line Segment
+				// Not currently relevant for BRDPin
+				current_pointer += *reinterpret_cast<uint32_t*>(&buf[current_pointer]) + 4; // Skip the block
+				break;
+			}
+			case 0x06: { // Labels/Part Names
+				// Not currently relevant for BRDPin
+				current_pointer += *reinterpret_cast<uint32_t*>(&buf[current_pointer]) + 4; // Skip the block
+				break;
+			}
+			case 0x09: { // Pins
+				pin_count += 1;
+				pin.side = BRDPinSide::Top;
+
+				// Block size
+				uint32_t pin_block_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+				uint32_t pin_block_end = current_pointer + pin_block_size + 4;
+				current_pointer += 4;
+				current_pointer += 4; // currently unknown
+				
+				
+				pin.pos.x = *reinterpret_cast<uint32_t*>(&buf[current_pointer]) / 10000;
+				current_pointer += 4;
+				pin.pos.y = *reinterpret_cast<uint32_t*>(&buf[current_pointer]) / 10000;
+				current_pointer += 4;
+				current_pointer += 8; // currently unknown
+
+				uint32_t pin_name_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+				current_pointer += 4;
+				std::string pin_name(reinterpret_cast<char*>(&buf[current_pointer]), pin_name_size);
+				
+				pin.name = strdup(pin_name.c_str());
+				pin.snum = strdup(pin_name.c_str());
+				current_pointer += pin_name_size;
+				current_pointer += 32;
+
+				uint32_t net_index = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+				current_pointer = pin_block_end;
+
+				std::string diode_reading = "";
+
+				std::string pin_net = net_dict[net_index];
+				
+				if (pin_net == "NC") {
+					pin.net = strdup("UNCONNECTED");
+				} else {
+					pin.net = strdup(pin_net.c_str());
+				}
+				
+				pin.part = parts.size() + 1;
+
+				if (diode_reading != "") {
```

Comment:
`diode_reading` is defined as empty string just above so this is not doing anything.